accounting_reporting_core/src/main/java/org/cardanofoundation/lob/app/accounting_reporting_core/repository/CustomTransactionRepositoryImpl.java

```diff
@@ -42,4 +48,123 @@ public List<TransactionEntity> findAllByStatus(String organisationId,
+
+    }
+
+    @Override
+    public List<Object[]> findAllReconciliationSpecialCount(ReconciliationFilterRequest body) {
+
+
+        String jpql = "SELECT count(rv) " +
+                "FROM accounting_reporting_core.ReconcilationEntity r " +
+                "JOIN r.violations rv " +
+                "LEFT JOIN accounting_reporting_core.TransactionEntity tr ON rv.transactionId = tr.id " +
+                "where r.id = tr.lastReconcilation.id or tr.id is null "
+                ;
+
+        String where = "";
+        if (!body.getReconciliationRejectionCode().isEmpty()) {
+            where += " and rv.rejectionCode in (" + body.getReconciliationRejectionCode().stream().map(Objects::toString).collect(Collectors.joining(",")) + ") ";
+        }
+        where += "group by rv.transactionId,tr.id,rv.amountLcySum,rv.rejectionCode,rv.sourceDiff,rv.transactionEntryDate,rv.transactionInternalNumber,rv.transactionType ";
+
+        return em.createQuery(jpql + where).getResultList();
+
+    }
+
+    @Override
+    public List<TransactionEntity> findAllReconciliation(ReconciliationFilterRequest body) {
+
+        switch (body.getFilter()) {
+            case ReconciliationFilterStatusRequest.RENCONCILED -> {
+                CriteriaBuilder builder = em.getCriteriaBuilder();
+                CriteriaQuery<TransactionEntity> criteriaQuery = builder.createQuery(TransactionEntity.class);
+                Root<TransactionEntity> rootEntry = criteriaQuery.from(TransactionEntity.class);
+
+                criteriaQuery.select(rootEntry);
+                criteriaQuery.where(builder.and(builder.equal(rootEntry.get("reconcilation").get("finalStatus"), ReconcilationCode.OK)));
+                TypedQuery<TransactionEntity> theQuery = em.createQuery(criteriaQuery);
+                theQuery.setMaxResults(body.getLimit());
+                if (null != body.getPage() && 0 < body.getPage()) {
+                    theQuery.setFirstResult(body.getPage() * body.getLimit());
+                }
+                return theQuery.getResultList();
+            }
+            case ReconciliationFilterStatusRequest.UNPROCESSED -> {
+                CriteriaBuilder builder = em.getCriteriaBuilder();
+                CriteriaQuery<TransactionEntity> criteriaQuery = builder.createQuery(TransactionEntity.class);
+                Root<TransactionEntity> rootEntry = criteriaQuery.from(TransactionEntity.class);
+
+                criteriaQuery.select(rootEntry);
+                criteriaQuery.where(builder.isNull(rootEntry.get("reconcilation").get("source")));
+                TypedQuery<TransactionEntity> theQuery = em.createQuery(criteriaQuery);
+                theQuery.setMaxResults(body.getLimit());
+                if (null != body.getPage() && 0 < body.getPage()) {
+                    theQuery.setFirstResult(body.getPage() * body.getLimit());
+                }
+                return theQuery.getResultList();
+            }
+        }
+
+        return null;
```

Comment:
cosmetics: replace with List.of() as we agreed to avoid return null anti-pattern.